wsd/DocumentBroker.cpp

```diff
@@ -274,6 +274,12 @@ void DocumentBroker::pollThread()
         _poll->poll(unloading ? SocketPoll::DefaultPollTimeoutMicroS / 16
                               : SocketPoll::DefaultPollTimeoutMicroS);
 
+        if (EnableExperimental && _stop)
```

Comment:
how is 'stop()' called by ->poll directly (or someone else?) and why does the normal processing take significant time? - surely it should not or everything will crawl ? =)

Reply:
`stop()` gets called through the Socket handler. What happens is that when the async uploading finishes, the handler (which is in DocumentBroker) is invoked and it figure that was the last upload before exiting, calls `stop()`.

Without checking for `_stop` after `poll()` we end up uploading yet again (when `always_save_on_exit` is set), which besides making testing harder (since depending on timing differences we might do an extra upload), it also uploads redundantly (this can't be helped on the storage side because `always_save_on_exit` forces the upload).